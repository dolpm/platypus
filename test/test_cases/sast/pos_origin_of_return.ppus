pipe inner |'c| [mic: &'c int] |> &'c int {       
    |> mic;
}

pipe main [] |> unit {
    int black = 1337;
    mut &int x = &black;
    {
        x = inner <| [&black];
    }
    |> ();
}