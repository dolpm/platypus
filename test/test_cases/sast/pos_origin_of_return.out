pipe inner |'c| [mic: &'c int] |> &'c int {
{(id: inner_with_args)
{(id: inner_wrapper)
  |> (&'c int : mic);
}
}
}

pipe main [] |> unit {
{(id: main_with_args)
{(id: main_wrapper)
  int black = (int : 1337);
  mut &'_ int x = (&'_ int : &(int : black));
{(id: 1)
  x = (&'_ int : inner <| [(&'_ int : &(int : black))]);
}
  |> (unit : ());
}
}
}
